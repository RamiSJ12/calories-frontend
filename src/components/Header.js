import {
  AppBar,
  IconButton,
  makeStyles,
  Toolbar,
  Typography,
} from '@material-ui/core';
import React from 'react';
import PropTypes from 'prop-types';
import ExitToAppIcon from '@material-ui/icons/ExitToApp';
import axios from 'axios';
import { useDispatch, useSelector } from 'react-redux';
import logout from '../actions/logout';

const useStyles = makeStyles({
  root: {
    justifyContent: 'space-between',
    flex: 2,
  },
  fontStyles: {
    fontWeight: 'bold',
    fontSize: '1.5em',
  },
  flex1: {
    flex: 1,
  },
  iconWrapper: {
    display: 'flex',
    justifyContent: 'end',
    flex: 1,
  },
  marginLeftAuto: {
    marginLeft: 'auto',
  },
});

const Header = ({ title, isLogged }) => {
  const dispatch = useDispatch();
  const classes = useStyles();
  const { uid, client, accessToken } = useSelector(state => state.sessionState);

  const logOut = (uid, accessToken, client) => {
    axios.delete('http://localhost:3000/auth/sign_out', {
      headers: {
        uid,
        client,
        'access-token': accessToken,
      },
    })
      .then(resp => {
        if (resp.status === 200) {
          dispatch(logout());
        }
      });
  };

  return (
    <AppBar color="primary" position="sticky">
      <Toolbar className={classes.root}>
        <div className={classes.flex1} />
        <Typography className={classes.fontStyles} component="h1" color="textPrimary" align="center">{title}</Typography>
        <div className={classes.iconWrapper}>
          { isLogged ? (
            <IconButton className={classes.marginLeftAuto} edge="end" onClick={() => logOut(uid, accessToken, client)}>
              <ExitToAppIcon />
            </IconButton>
          ) : ''}
        </div>
      </Toolbar>
    </AppBar>
  );
};

Header.propTypes = {
  title: PropTypes.string.isRequired,
  isLogged: PropTypes.bool,
};

Header.defaultProps = {
  isLogged: true,
};

export default Header;
