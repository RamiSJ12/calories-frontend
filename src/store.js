import { createReduxStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from './reducers/rootReducer';

const store = createReduxStore(rootReducer, applyMiddleware(thunk));

export default store;
