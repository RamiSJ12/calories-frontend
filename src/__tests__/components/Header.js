import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Header from '../../components/Header';
import store from '../../store';

describe('Header component', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Header title="Home" />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
