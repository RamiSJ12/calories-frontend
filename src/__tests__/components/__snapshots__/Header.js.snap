// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular makeStyles-root-1 MuiToolbar-gutters"
  >
    <div
      className="makeStyles-flex1-3"
    />
    <h1
      className="MuiTypography-root makeStyles-fontStyles-2 MuiTypography-body1 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
    >
      Home
    </h1>
    <div
      className="makeStyles-iconWrapper-4"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root makeStyles-marginLeftAuto-5 MuiIconButton-edgeEnd"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</header>
`;
